frdm_kw41z: Add pyocd target

MaureenHelm opened the pull request:
Adds the pyocd target name for the frdm_kw41z board, but does not change
the default flash/debug scripts from jlink to pyocd. pyocd has not yet
tagged a release with kw41z support, so to use it one must build pyocd
from source based on the current master branch (f21d43d).

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
frdm_kw41z: Add pyocd target

Adds the pyocd target name for the frdm_kw41z board, but does not change
the default flash/debug scripts from jlink to pyocd. pyocd has not yet
tagged a release with kw41z support, so to use it one must build pyocd
from source based on the current master branch (f21d43d).

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

galak commented on boards/arm/frdm_kw41z/Makefile.board:
```diff
@@ -2,4 +2,6 @@ DEBUG_SCRIPT = jlink.sh
 
```
Shouldn't we than do:

DEBUG_SCRIPT ?= jlink.sh
FLASH_SCRIPT ?= pyocd.sh

MaureenHelm replied:
jlink.sh and pyocd.sh require different OpenSDA firmware, so this would mean you have to update the firmware each time you switch between `make flash` and `make debug`. I think we should pick a preferred firmware for each board, then set the DEBUG_SCRIPT and FLASH_SCRIPT defaults accordingly. In the case of jlink firmware, that means no default FLASH_SCRIPT.

galak replied:
I like the idea, maybe something like OPENSDA_FW_TYPE={JLINK, MBED, ...} and use that to set things

dbkinder replied:
It would be good to update the documentation to talk about different OpenSDA firmware needed for jlink.sh and pyocd.sh

MaureenHelm replied:
The docs already do talk about the different OpenSDA firmwares. See doc/tools/opensda.rst

galak reviewed (changes requested):
Should we set the FLASH_SCRIPT & allow the DEBUG_SCRIPT to be overwritten by the user

MaureenHelm committed:
frdm_kw41z: Select flash/debug scripts based on OpenSDA firmware

Adds a new variable to define which OpenSDA firmware is on the board,
and selects the default flash/debug scripts accordingly.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

galak reviewed (commented):
Should we update the docs to reflect supporting jlink & pyocd?

What about the other NXP boards?

dbkinder commented:
Yes, please.  More documentation (that's up to date) is good.

MaureenHelm commented:
> Should we update the docs to reflect supporting jlink & pyocd?

Which docs? The OpenSDA doc or the individual board docs?

> What about the other NXP boards?

If you're good with this change, I'll update the other boards

galak commented:
The individual board docs.  I'm good with the changes.

galak commented:
But something in the opensda doc would probably be good as well.

MaureenHelm committed:
frdm_k64f: Select flash/debug scripts based on OpenSDA firmware

Adds a new variable to define which OpenSDA firmware is on the board,
and selects the default flash/debug scripts accordingly.

Sets the default to daplink since the flashing and debugging examples in
the board document were written to use pyocd.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
frdm_kl25z: Select flash/debug scripts based on OpenSDA firmware

Adds a new variable to define which OpenSDA firmware is on the board,
and selects the default flash/debug scripts accordingly.

Sets the default to daplink since the flashing and debugging examples in
the board document were written to use pyocd.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
hexiwear_k64: Select flash/debug scripts based on OpenSDA firmware

Adds a new variable to define which OpenSDA firmware is on the board,
and selects the default flash/debug scripts accordingly.

Sets the default to daplink since the flashing and debugging examples in
the board document were written to use pyocd.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
hexiwear_kw40z: Select flash/debug scripts based on OpenSDA firmware

Adds a new variable to define which OpenSDA firmware is on the board,
and selects the default flash/debug scripts accordingly.

Sets the default to jlink since the flashing and debugging examples in the
board document were written to use jlink.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
doc: Update OpenSDA doc to reflect new OPENSDA_FW variable

Kinetis boards now have a makefile variable to select the flash/debug
scripts based on the OpenSDA firmware. Update the general OpenSDA
documentation accordingly.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm commented:
Rebased to latest arm branch, updated all the other NXP boards and docs

galak committed:
Merge branch 'arm' into pyocd-kw41z

galak committed:
Merge branch 'arm' into pyocd-kw41z
